### PR TITLE
Critical Alerts detection - Microsoft Defender

### PR DESCRIPTION
**Pull request by gowthamarajr**
### Details

Detection -  Detect Critical Alerts from Security Tools
Story - Critical Alerts

### Checklist

- [ ] Validate name matches `<platform>_<mitre att&ck technique>_<short description>` nomenclature
- [ ] [CI/CD](https://github.com/splunk/security_content/actions) jobs passed ✔️ 
- [ ] Validated SPL logic.
- [ ] Validated tags, description, and how to implement.
- [ ] Verified references match analytic.
- [ ] Confirm updates to lookups are handled properly.

### Notes For Submitters and Reviewers

- If you're submitting a PR from a fork, ensuring the box to allow updates from maintainers is checked will help speed up the process of getting it merged.
- Checking the output of the `build` CI job when it fails will likely show an error about what is failing. You may have a very descriptive error of the specific field(s) in the specific file(s) that is causing an issue. In some cases, its also possible there is an issue with the YAML. Many of these can be caught with the pre-commit hooks if you set them up. These errors will be less descriptive as to what exactly is wrong, but will give you a column and row position in a specific file where the YAML processing breaks. If you're having trouble with this, feel free to add a comment to your PR tagging one of the maintainers and we'll be happy to help troubleshoot it.
- Updates to existing lookup files can be tricky, because of how Splunk handles application updates and the differences between existing lookup files being updated vs new lookups. You can read more [here](https://docs.splunk.com/Documentation/SplunkCloud/8.2.2203/Admin/PrivateApps#Manage_lookups_in_Splunk_Cloud_Platform) but the short version is that any changes to lookup files need to bump the datestamp in the lookup CSV filename, and the reference to it in the YAML needs to be updated.

### CODE REVIEW

File: detections/endpoint/critical_alerts_to_risk_index.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Critical Alerts To Risk Index
```
Comment (patel-bhavin) [inline detections/endpoint/critical_alerts_to_risk_index.yml]: consider naming this search to Detect Critical Alerts from Security Tools

File: detections/endpoint/critical_alerts_to_risk_index.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  asset_type: Endpoint
+  atomic_guid: []
+  confidence: 90
+  impact: 90
+  message: $severity$ alert from $dest$ for $app$ with signature $signature$ and description $description$
+  mitre_attack_id:
+  - T1484
+  observable:
+  - name: dest
+    type: Other
```
Comment (patel-bhavin) [inline detections/endpoint/critical_alerts_to_risk_index.yml]: should be Endpoint

File: stories/critical_alerts.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Critical Alerts
+id: bc7056a5-c2b0-4b83-93ce-5f31739305c8
+date: '2024-06-21'
+author: Gowthamaraj Rajendran, Patrick Bareiss, Splunk
+description: Rule for sending alerts to risk index.
```
Comment (patel-bhavin) [inline stories/critical_alerts.yml]: can we improve the text here?

File: detections/endpoint/critical_alerts_to_risk_index.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+status: production
+type: TTP
+data_source: []
+description: The primary objective of this rule is to integrate and assess critical alerts from Endpoint, DLP, and firewall sources within the splunk system. By correlating these alerts and incorporating MITRE annotations, the rule provides a comprehensive view of customer risk. It triggers an alert when critical alerts from these categories are detected, preserving the originating source and assigning risk scores. The rule helps security analysts better understand potential threats, enabling timely and effective responses to mitigate risks. The results are collected in the risk index for continuous monitoring and analysis.
+search: '| tstats `security_content_summariesonly` count min(_time) as firstTime max(_time) as lastTime from datamodel=Alerts where Alerts.severity IN ("high","critical") by Alerts.signature Alerts.app, Alerts.severity, Alerts.description, source, Alerts.id, Alerts.dest 
+  | `drop_dm_object_name("Alerts")`
+  | `critical_alerts_to_risk_index_filter`'
+how_to_implement: In order to properly run this search, Splunk needs to ingest data from other security products.
```
Comment (patel-bhavin) [inline detections/endpoint/critical_alerts_to_risk_index.yml]: add more detail here

**Comment by patel-bhavin** [comment]
can you test this locally to figure out why is it failing  unit- testing?

File: detections/endpoint/critical_alerts_to_risk_index.yml (hunk 1 above)
Comment (gowthamarajr) [reply detections/endpoint/critical_alerts_to_risk_index.yml]: Done

File: detections/endpoint/detect_critical_alerts_from_security_tools.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  asset_type: Endpoint
+  atomic_guid: []
+  confidence: 90
+  impact: 90
+  message: $severity$ alert from $dest$ for $app$ with signature $signature$ and description $description$
+  mitre_attack_id:
+  - T1484
+  observable:
+  - name: dest
+    type: Other
```
Comment (patel-bhavin) [inline detections/endpoint/detect_critical_alerts_from_security_tools.yml]: update : Other to Endpoint

File: stories/critical_alerts.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+name: Critical Alerts
+id: bc7056a5-c2b0-4b83-93ce-5f31739305c8
+date: '2024-06-21'
+author: Gowthamaraj Rajendran, Patrick Bareiss, Splunk
+description: The rules are designed to integrate and assess critical alerts from Endpoint, DLP, and firewall sources within Splunk. By correlating these alerts and enriching them with MITRE ATT&CK annotations, it offers a nuanced perspective on alerts risk profiles. This rule actively monitors for critical alerts, and upon detection, it triggers an alert that not only preserves the original source information but also assigns a quantified risk score.
+narrative: This story has rules that integrates and assesses critical alerts from Endpoint, DLP, and firewall sources in Splunk. By correlating alerts and adding MITRE annotations, it provides a comprehensive view of customer risk. It triggers an alert when critical alerts are detected, preserving the source and assigning risk scores. This helps security analysts understand threats and respond effectively.
+references:
```
Comment (patel-bhavin) [inline stories/critical_alerts.yml:7]: can we add some references

File: detections/endpoint/detect_critical_alerts_from_security_tools.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+search: '| tstats `security_content_summariesonly` count min(_time) as firstTime max(_time) as lastTime from datamodel=Alerts where Alerts.severity IN ("high","critical") by Alerts.signature Alerts.app, Alerts.severity, Alerts.description, source, Alerts.id, Alerts.dest 
+  | `drop_dm_object_name("Alerts")` | `security_content_ctime(firstTime)` | `security_content_ctime(lastTime)`
+  | `detect_critical_alerts_from_security_tools_filter`'
+how_to_implement: In order to properly run this search, Splunk needs to ingest data from other security products such as crowdstrike, microsoft defender, or carbon black. Once ingested, the fields should be mapped to the Alerts data model. Make sure to apply transformation on the data if necessary. 
+known_false_positives: False positives may vary by endpoint protection tool; monitor and adjust the risk scores as needed.
+references:
+  - https://attack.mitre.org/tactics/
```
Comment (patel-bhavin) [inline detections/endpoint/detect_critical_alerts_from_security_tools.yml]: can we improve references

File: detections/endpoint/detect_critical_alerts_from_security_tools.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
+name: Detect Critical Alerts from Security Tools
+id: 483e8a68-f2f7-45be-8fc9-bf725f0e22fd
+version: 1
+date: '2024-06-04'
+author: Gowthamaraj Rajendran, Patrick Bareiss, Splunk
+status: production
+type: TTP
+data_source: ["windows defender logs"]
```
Comment (patel-bhavin) [inline detections/endpoint/detect_critical_alerts_from_security_tools.yml]: lets add a data source file for this data in the data sources folder and use that name in here

File: contentctl.yml [hunk 9]
```diff
@@ -176,6 +176,12 @@ apps:
   version: 5.4.1
   description: description of app
   hardcoded_path: https://attack-range-appbinaries.s3.us-west-2.amazonaws.com/Latest/splunk-machine-learning-toolkit_541.tgz
+- uid: 5518
+  title: Splunk add on for Microsoft Defender Advanced Hunting
+  appid: SPLUNK_ADD_ON_FOR_MICROSOFT_DEFENDER_ADVANCED_HUNTING
+  version: 1.3.7
+  description: description of app
+  hardcoded_path: https://attack-range-appbinaries.s3.us-west-2.amazonaws.com/microsoft-defender-advanced-hunting-add-on-for-splunk_141.tgz
```
Comment (ljstella) [inline contentctl.yml:184]: Which app version are we using here? the `version` string says one thing, but the filename at the URL indicates it might be something else.

File: data_sources/windows_defender_alerts.yml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+date: '2024-09-24'
+author: Gowthamaraj Rajendran
+description: Data source object for Windows Defender alerts
+source: eventhub://windowsdefenderlogs
+sourcetype: mscs:azure:eventhub:defender:advancedhunting
+separator: AlertId
+supported_TA:
+  - name: Splunk add on for Microsoft Defender Advanced Hunting
+    url: https://splunkbase.splunk.com/app/5518
+    version: 1.3.7
```
Comment (ljstella) [inline data_sources/windows_defender_alerts.yml]: See previous comment about version

File: detections/endpoint/detect_critical_alerts_from_security_tools.yml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  - https://docs.splunk.com/Documentation/CIM/5.3.2/User/Alerts
+  - https://learn.microsoft.com/en-us/defender-endpoint/api/raw-data-export-event-hub
+tags:
+  analytic_story:
+  - Critical Alerts
+  asset_type: Endpoint
+  atomic_guid: []
+  confidence: 90
+  impact: 90
+  message: $severity$ alert from dest - [$dest$] for $app$ with signature [$signature$] from vendor source - [$source$]
```
Comment (ljstella) [inline detections/endpoint/detect_critical_alerts_from_security_tools.yml]: This feels a bit wordy the way we duplicate things, and especially since signature can be a bit lengthy-

Suggest:

```
$severity$ alert for $dest$ from $source$ - $signature$
```
Even that, the `$signature$` on the end might not even be necessary. 

Additionally, I think we'd want to triple check across more than one TA as to how they usually set the `source` and `signature`, as some vendors might use it in one way while another could do something entirely different- think about the difference between one vendor putting a short thing that is incredibly information-dense like `Sanesecurity.Malware.30675.MSC.JSeval.UNOFFICIAL` while another vendor would just say "ML Detection" or "GuardDuty Detection"- and how that looks across these.

**Review by ljstella — COMMENTED** [review]
Couple small things

File: detections/endpoint/detect_critical_alerts_from_security_tools.yml (hunk 11 above)
Comment (patel-bhavin) [reply detections/endpoint/detect_critical_alerts_from_security_tools.yml]: i think this is good : $severity$ alert for $dest$ from $source$ - $signature$

**Review by ljstella — APPROVED** [review]
LGTM

:shipit:

**Comment by patel-bhavin** [comment]
Known CI Error . the other jobs look good!
